syntax = "proto3";
package AccountMessage;

// proto_id = sid + cid;
enum proto_ids
{
    option allow_alias = true;
    EMPTY_PROTO = 0;

    // service id
    ACCOUNT_SERVICE = 0x0001;

    // command id
    LOGIN = 0x0001;
    REGISTER = 0x0002;
}

message LoginRequest
{
    string user_name = 1;
    string password = 2;
}

message LoginResponse
{
    int32 result = 1;
    int32 uid = 2;
}

service Account
{
    rpc login (LoginRequest) returns (LoginResponse) { }
}